app/questionnaire/router.py

```diff
@@ -210,19 +206,23 @@ def is_path_complete(self, routing_path):
             return True
         return False
 
-    def get_section_return_location_when_section_complete(
-        self, routing_path
-    ) -> Location:
+    def get_section_return_location_when_section_complete(self, routing_path):
 
-        section_summary_location = self._get_location_of_section_summary(routing_path)
-        if section_summary_location:
-            return section_summary_location
+        last_block_id = routing_path[-1]
+        block = self._schema.get_block(last_block_id)
 
+        if block["type"] == "SectionSummary":
```

Comment:
This also needs to check for `ListCollectorSummary` as currently the census schemas who lives here doesn't go back to the summary from the hub when it's complete.